### DISCUSSION TITLE
Trouble uploading file using FileSystem library

### DISCUSSION THREAD

**Discussion by Zakisb** [expo/expo#18017] [Archived: Questions]
So i'm building a marketplace mobile app using, expo that let users upload products to the marketplace. I'm having difficult times using the expo FileSystem.FileSystemUploadType to pick the image and send to the backend. here's my front-end code 
```js
const handleSubmit = async (listing, { resetForm }) => {
    
    const data = new FormData();
    data.append("images", listing.images[0]);
    data.append("description", "Good product");

    console.log( listing.images[0])
   // it does console the file uri
   // file:///Users/zakisb/Library/Developer/CoreSimulator/Devices/83E12EA5-E8FA-4850-82C1-84021B25450D/data/Containers/Data/Application/6873BF40-26E4-4BD3-834D-F6772448C004/Library/Caches/ExponentExperienceData/%2540anonymous%252Flokazz_app2-5f4724db-b9d7-45aa-a8ca-ac5acf2f4780/ImagePicker/B72CF40C-EC27-430E-B1F8-B983C0ACF2FB.jpg
   
  // i tried this solution first and worked perfectly. but it does upload the image only and i want to send the formdata object
 const response = await FileSystem.uploadAsync(
      "http://192.168.43.8:5000/products/addProduct",
      listing.images[0],
      {
        fieldName: "images",
        uploadType: FileSystem.FileSystemUploadType.MULTIPART,
      }
    );

   // so i tried this but i get an error
    const response = await FileSystem.FileSystemUploadType.MULTIPART(
      "http://192.168.43.8:5000/products/addProduct",
      data
    );
    
  };
````
My backend works perfectly. i tried the api call using insomnia and postman the file gets uploaded successfully to the folder. but using expo methods i get nothing.

```js
const storage = multer.diskStorage({
  destination: function (req, file, cb) {
    cb(null, "./images");
  },
  filename: function (req, file, cb) {
    cb(null, Date.now() + file.originalname);
  },

router.post(
  "/products/addProduct",
  upload.single("images"),
  async (req, res) => {
    console.log(req.body);
    try {
      res.send(req.body);
    } catch (err) {
      res.send("Error " + err);
    }
  }
);
});


```
Please help !

**Comment by barthap** [comment]
```js
// so i tried this but i get an error
    const response = await FileSystem.FileSystemUploadType.MULTIPART(
      "http://192.168.43.8:5000/products/addProduct",
      data
    );
```
You are trying to call enum value as a function 😉 

Long time ago I was doing some upload using `expo-file-system` - you can look at https://github.com/barthap/MemeBrowser/blob/2458d50bb554d9cf27bfa28842f8602e5cc11456/src/model/ocrApi.ts

Also check out the commented-out code - it should be working now. I use the `parameters` field to send additional data to the server. Something like:
```js
 const response = await FileSystem.uploadAsync(
      "http://192.168.43.8:5000/products/addProduct",
      listing.images[0],
      {
        fieldName: "images",
        uploadType: FileSystem.FileSystemUploadType.MULTIPART,
        parameters: {
          description: "Good product"
        }
      }
    );
```

**Comment by Zakisb** [comment]
THANKS A LOT ! the image did get uploaded using the fetch method, but the problem when i receive it in node.js it says : the file is not displayed in the editor because it is either binary or uses unsupported ext encoding , can u please help me out ? i'm getting closer !!

**Reply by barthap** [reply]
try uploading a text file first, for example created by `FileSystem.writeAsStringAsync` - it is easier to debug if it gets corrupted during the upload process

**Reply by Zakisb** [reply]
what do you mean? sorry i 'm still a beginner  so i think more details is necessary please !

**Reply by Zakisb** [reply]
using a piece of a code would be better

**Reply by Rise-admin1** [reply]
were you able to fix this issue?
